src/index.ts

```diff
@@ -1,64 +1,160 @@
+					`Duplicate validFrom ${validFrom} for feature ${slug}`,
+				);
+			}
+
+			versions.add(definition.version);
+			validFroms.add(validFrom);
+		}
 	}
-	return pricing;
+}
+
+/**
+ * Sort credit pricing definitions by validFrom date.
+ * Sorts from latest to oldest.
+ * @param credits - credit pricing definitions
+ * @returns sorted credit pricing definitions
+ *
+ * @example
+ * sortCredits(CREDITS);
+ */
+function sortCredits(credits: Credits): Credits {
+	const sortedCredits: Credits = {};
+	for (const [slug, definitions] of Object.entries(credits)) {
+		sortedCredits[slug] = definitions.sort((a, b) => {
+			return b.validFrom.getTime() - a.validFrom.getTime();
+		});
+	}
+	return sortedCredits;
 }
 
 export class CreditPricing {
-	public credits: { [key: string]: Credit };
+	public credits: { [slug: string]: Credit[] };
+	private target: 'current' | 'latest' | number | Date;
 
-	public constructor(credits = CREDITS) {
-		this.credits = credits;
+	public constructor(options: Options = {}) {
+		// Sort and then validate credit pricing definitions.
+		this.credits = sortCredits(options.credits ?? CREDITS);
+		validateCredits(this.credits);
+
+		// Allow consumers to target one of the following:
+		//  'current' - the most recent valid version (default)
+		//  'latest' - the latest version, regardless of validity
+		//  number - a specific version
+		//  Date - the most recent valid version up to a given date
+		this.target = options.target ?? 'current';
 	}
 
 	/**
-	 * Checks if a given feature slug has a credit pricing definition.
-	 * @param featureSlug - feature slug to check
-	 * @returns boolean denoting if feature has a credit pricing definition
+	 * Gets and returns pricing for a given feature.
+	 * @param featureSlug - feature slug
+	 * @returns credit pricing definition
+	 *
+	 * @example
+	 * getDefinition('device:microservices');
 	 */
-	public exists(featureSlug: string): boolean {
-		return Object.keys(this.credits).includes(featureSlug);
+	public getDefinition(featureSlug: string): Credit | undefined {
+		if (this.credits[featureSlug] == null) {
+			throw new InvalidParametersError(
+				`Feature ${featureSlug} not supported for credits`,
+			);
+		}
+
+		let definition: Credit | undefined;
+
+		// Return latest version of a feature's pricing definition set.
+		// This ignores the validFrom date, meaning it can/will return
+		// definitions that are scheduled to be valid in the future.
+		if (this.target === 'latest') {
+			return this.credits[featureSlug][0];
+		}
+
+		// Return exact version of a feature's pricing definition set.
+		if (typeof this.target === 'number') {
+			definition = this.credits[featureSlug].find((credit) => {
+				return credit.version === this.target;
+			});
+			return definition;
+		}
+
+		// Return version of a feature's pricing definition set that is
+		// valid up to a given date.
+		if (this.target instanceof Date) {
+			[definition] = this.credits[featureSlug].filter((def) => {
```

Comment:
This can be a `.find` as well, right?

Reply:
Yeah, that's a leftover from before when `filter` was the better option. 👍